airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -561,8 +561,11 @@ definitions:
         enum: [DatetimeBasedCursor]
       cursor_field:
         title: Cursor Field
-        description: The location of the value on a record that will be used as a bookmark during sync.
+        description: The location of the value on a record that will be used as a bookmark during sync. To ensure no data loss, the API must return records in ascending order based on the cursor field. Nested fields are not supported, so the field must be at the top level of the record. You can use a combination of Add Field and Remove Field transformations to move the nested field to the top.
         type: string
+        interpolation_context:
+          - config
+          - stream_state
```

Comment:
One more question I just thought of: why doesn't this interpolation context have access to the records/response? It seems like if it did, the nested cursor field issue could be solved by accessing that variable in the interpolation

Reply:
the cursor field is the path on the record. This won't work if the path is in the response, but not in the record.

We could update it to be a list to support nested fields, but normalization won't support nested cursor fields so there's not much value anyway.

(I'm not sure in what scenario the stream_state would be needed here - it would probably make more sense for the cursor_field to be static or at most, defined in the config)